From 3938b7be5ebb91bb0c19433026873ad0abfb7b3c Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Tue, 24 Jun 2014 23:15:29 +0300
Subject: [PATCH] Use INSTALLDIRS=vendor for the Perl bindings, as per the
 Debian Perl policy

Debian packaged Perl modules should be installed into the vendor
directories (currently /usr/lib/perl5 and /usr/share/perl5), but the
upstream default puts the into the 'site' ones (modified with the
PREFIX setting.) Explicitly using INSTALLDIRS=vendor simplifies
the install rules and removes the need for PREFIX, which has
been deprecated in ExtUtils::MakeMaker for a long time.
---
 bindings/perl/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bindings/perl/Makefile.am b/bindings/perl/Makefile.am
index 67e57a6..8f30a45 100644
--- a/bindings/perl/Makefile.am
+++ b/bindings/perl/Makefile.am
@@ -42,7 +42,7 @@ Makefile.perl: $(srcdir)/Makefile.PL Lasso.xs Lasso.pm
 		done; \
 		chmod -R u+rwX $(TOCOPY); \
 	fi;
-	$(AM_V_SUBMAKE) $(PERL) Makefile.PL INSTALLDIRS=vendor CCFLAGS="$(LASSO_XS_CFLAGS)" INC="-I. -I$(top_srcdir) -I$(srcdir)" LIBS="`$(top_builddir)/lasso-src-config --libs`" OPTIMIZE="-g" $(AM_V_SUBMAKE_POSTFIX)
+	$(AM_V_SUBMAKE) $(PERL) Makefile.PL INSTALLDIRS=vendor CCFLAGS="$(LASSO_XS_CFLAGS)" INC="-I. -I$(top_srcdir) -I$(srcdir)" LIBS="`$(top_builddir)/lasso-src-config --libs`" OPTIMIZE="-g" $(AM_V_SUBMAKE_POSTFIX)
 
 Lasso.xs Lasso.pm: lang.py typemap-in typemap-out
 	$(AM_V_GEN) $(PYTHON) $(top_srcdir)/bindings/bindings.py -l perl --src-dir=$(top_srcdir)/lasso/ $(EXTRA_ARGS)
@@ -59,7 +59,7 @@ clean-local:
 	-@if [ -f Makefile.perl ]; then make -f Makefile.perl clean; fi; rm -f Makefile.perl
 
 install-exec-local: Makefile.perl
-	make -f Makefile.perl install DESTDIR=$(DESTDIR) PREFIX=$(prefix)
+	make -f Makefile.perl install DESTDIR=$(DESTDIR)
 
 uninstall-local:
 	-@rm -rf `find $(DESTDIR)$(prefix) -name Lasso.pm -or -name Lasso.so -or -name Lasso.bs -or -name .packlist -or -name perllocal.pod`
-- 
2.0.0

